main.cpp

```diff
@@ -14,28 +52,68 @@ class main_window : public vsite::nwp::window
 
 int main_window::on_create(CREATESTRUCT* pcs)
 {
-	// TODO: create all child windows
-	// TODO: disable "Remove" button
+	// : create all child windows
+	char remove_text[128];
+	::LoadString(0, IDC_REMOVE, remove_text, sizeof(remove_text));
+	char add_text[128];
+	::LoadString(0, IDC_ADD, add_text, sizeof(add_text));
+	
+	button add;
+	button remove;
+	edit edit;
+	list_box list_box;
+
+	add.create(*this,WS_CHILD| WS_VISIBLE,add_text, IDC_ADD, 100, 50, 80, 24);
+	remove.create(*this,WS_CHILD | WS_VISIBLE,remove_text,IDC_REMOVE, 100, 80, 80, 24);
+	edit.create(*this, WS_CHILD | WS_VISIBLE | WS_BORDER, 0, IDC_EDIT, 100, 20, 80, 24);
+	list_box.create(*this, WS_CHILD | WS_VISIBLE | WS_BORDER, 0, IDC_LB, 20, 20, 70, 90);
+	
+	EnableWindow(remove, false);
+	
+	// : disable "Remove" button
 	return 0;
 }
 
+
 void main_window::on_command(int id){
 	switch(id){
+		int textLength;
+		char text[256];
+		HWND hwListBox;
+		HWND hwEdit;
 		case ID_FILE_EXIT:
-			// TODO: close main window
+			DestroyWindow(*this);
+			// : close main window
 			break;
 		case ID_HELP_ABOUT:
-			// TODO: show dialog with text
+			// : show dialog with text
+			DialogBox(GetModuleHandle(NULL), MAKEINTRESOURCE(ID_HELP_ABOUT), 0, AboutDialogProc);
 			break;
 		case IDC_ADD:
-			// TODO: get text from edit control
-			// TODO: add string to listbox control
-			// TODO: enable "Remove" button
+			textLength = GetWindowTextLength(GetDlgItem(*this, IDC_EDIT)) + 1;
+			GetWindowText(GetDlgItem(*this, IDC_EDIT), text, textLength);
+			hwListBox = GetDlgItem(*this, IDC_LB);
+			hwEdit = GetDlgItem(*this, IDC_EDIT);
+			SendMessage(hwListBox, LB_ADDSTRING, 0, (LPARAM)text);

```

Comment:
Zašto ne `SendDlgItemMessage` kako piše u uputstvima (pa izbjegnete `GetDlgItem`)?
(i niže, na par mjesta)